### PR TITLE
Add email parameter to all network requests to include email in server logs

### PR DESCRIPTION
**Pull request by Jag96**
### Details
This PR updates our API requests to include the `email` param, which will set the email in our server logs. I've removed the logic from `Log.js` and updated Network.js to include the email param, so every case should be covered. I've also added the email param to the initial Authenticate/CreateLogin calls, since the ONYX key for session isn't updated until after the successful CreateLogin.

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/145069

### Tests
1. Run the app on web and tail the logs, and open the Network console
2. Sign into the app and send some messages, confirm you see `email` is included as a `form data` parameter on the Network requests
3. In the logs, confirm you see the testing email next to the requestID
```
BEFORE
2021-02-13T00:32:45.017706+00:00 expensidev php-fpm: XIeVuz /api.php android:joe@expensify.com !https://expensify.cash/! ?api? [info] UserAPI - authenticate called ~~ partnerName: 'android' partnerUserID: 'joe@expensify.com' useExpensifyLogin: '1'
2021-02-13T00:32:45.041184+00:00 expensidev bedrock: XIeVuz (Authenticate.cpp:68) peek [worker2] [info] Attempting authenticate for partnerUserID=joe@expensify.com, partnerName=android

AFTER
2021-02-13T00:34:12.925616+00:00 expensidev php-fpm: 2TsFNX /api.php joe@expensify.com !https://expensify.cash/! ?api? [info] UserAPI - authenticate called ~~ partnerName: 'android' partnerUserID: 'joe@expensify.com' useExpensifyLogin: '1'
2021-02-13T00:34:12.941117+00:00 expensidev bedrock: 2TsFNX joe@expensify.com (Authenticate.cpp:68) peek [worker2] [info] Attempting authenticate for partnerUserID=joe@expensify.com, partnerName=android
```
4. Clear the logs and re-test on Desktop, iOS, and Android. Confirm the email continues to show up in the logs.
5. Kill the wifi on your laptop (I was using ngrok) and confirm the network requests begin to fail.
6. For the failed network requests, confirm the `email` param is shown in `form data` in the Network tab of the console
7. Re-enable wifi, confirm you can send messages, sign out, log in, and no new error messages appear in the console

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

### Screenshots
n/a

### CODE REVIEW

File: src/libs/Network.js [hunk 1]
```diff
@@ -53,9 +67,14 @@ function processNetworkRequestQueue() {
...
+        const requestData = queuedRequest.data;
+        if (email) {
+            requestData.email = email;
+        }
+
         const finalParameters = _.isFunction(enhanceParameters)
-            ? enhanceParameters(queuedRequest.command, queuedRequest.data)
-            : queuedRequest.data;
+            ? enhanceParameters(queuedRequest.command, requestData)
+            : requestData;
```
Comment (marcaaron) [inline src/libs/Network.js:77]: NAB, might make more sense to add the email to the `finalParameters` if we have one. Then we do not need `requestData`.

**Review by marcaaron — APPROVED** [review]
Pumped for this, thanks @Jag96 

I have a pretty small NAB so will approve and leave the choice to merge in your hands

File: src/libs/API.js [hunk 2]
```diff
@@ -187,6 +187,7 @@ function getAuthToken() {
  * @param {String} parameters.partnerUserID
  * @param {String} parameters.partnerUserSecret
  * @param {String} [parameters.twoFactorAuthCode]
+ * @param {String} [parameters.email]
```
Comment (deetergp) [inline src/libs/API.js:190]: NAB: Why are some of these in brackets and others are not?

**Review by deetergp — APPROVED** [review]

File: src/libs/API.js (hunk 2 above)
Comment (marcaaron) [reply src/libs/API.js:190]: Brackets mean the parameter is optional.

Comment (deetergp) [reply src/libs/API.js:190]: TIL, thanks!

**Pull request merged**